pkg/steps/template.go

```diff
@@ -48,6 +57,52 @@ type templateExecutionStep struct {
 	subTests []*junit.TestCase
 }
 
+func injectResourcesToTemplatePods(template *templateapi.Template, resources api.ResourceConfiguration) error {
+	if resources != nil {
+		containerResources, err := resourcesFor(resources.RequirementsForStep(template.Name))
+		if err != nil {
+			return fmt.Errorf("unable to calculate resources for %s: %s", template.Name, err)
+		}
+
+		for i, object := range template.Objects {
+			var pod *coreapi.Pod
+			if err := yaml.Unmarshal(object.Raw, &pod); err != nil {
+				return fmt.Errorf("couldn't unmashal pod %s: %v", pod.Name, err)
+			}
+
+			if pod != nil {
+				for index, container := range pod.Spec.Containers {
+					if container.Name == TestContainerName {
+						pod.Spec.Containers[index].Resources = containerResources
+						break
+					}
+				}
+
+				corev1Scheme := runtime.NewScheme()
+				utilruntime.Must(coreapi.AddToScheme(corev1Scheme))
+				corev1Codec := serializer.NewCodecFactory(corev1Scheme).LegacyCodec(coreapi.SchemeGroupVersion)
+
+				template.Objects[i] = runtime.RawExtension{
+					Raw:    []byte(runtime.EncodeOrDie(corev1Codec, pod)),
+					Object: pod.DeepCopyObject(),
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func injectLabelsToTemplatePod(jobSpec *api.JobSpec, template *templateapi.Template) {
```

Comment:
nit: you are injecting to the template labels, not to a pod (directly) so the function name is misleading

Reply:
addressed